core/capabilities/ccip/ocrimpls/svm_contract_transmitter_factory.go

```diff
@@ -59,11 +59,9 @@ var SVMExecCalldataFunc = func(
 	extraDataCodec *ccipcommon.ExtraDataCodec,
 ) (contract string, method string, args any, err error) {
 	var info ccipocr3.ExecuteReportInfo
-	if len(report.Info) != 0 {
-		info, err = ccipocr3.DecodeExecuteReportInfo(report.Info)
-		if err != nil {
-			return "", "", nil, fmt.Errorf("failed to decode execute report info: %w", err)
-		}
+	info, err = ccipocr3.DecodeExecuteReportInfo(report.Info)
+	if err != nil {
+		return "", "", nil, fmt.Errorf("failed to decode execute report info: %w", err)
```

Comment:
What if there's no info? You can fix the ambiguous `decodeExtraData` call by moving it into the if block:
```
	var info ccipocr3.ExecuteReportInfo
	var extraDataDecoded ccipcommon.ExtraDataDecoded
	if len(report.Info) != 0 {
		info, err = ccipocr3.DecodeExecuteReportInfo(report.Info)
		if err != nil {
			return "", "", nil, fmt.Errorf("failed to decode execute report info: %w", err)
		}

		if extraDataCodec != nil {
			extraDataDecoded, err = decodeExecData(info, extraDataCodec)
			if err != nil {
				return "", "", nil, fmt.Errorf("failed to decode extra data: %w", err)
			}
		}
	}
```

Reply:
Hmm, you pointed to another issue.
Originally, the ticket just said that DecodeExecuteReportInfo() was already handling the case of report.Info == 0. So remove the redundant check outside.

However, you're pointing that if it was 0, then decodeExecData() would error out.

Your suggested fix would silently return success with the SVMExecCallArgs having an empty report.
Is that what is really desired? Or we want to keep erroring here for empty reports?

Reply:
I guess I don't feel too strongly about it either way. But this call used to panic in `chainlink-ccip`, so I don't mind the defensive check.